src/homepageExperience/components/steps/cli/InitializeTokenClient.tsx

```diff
@@ -0,0 +1,137 @@
+// Libraries
+import React, {FC, useEffect, useMemo} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+
+// Actions
+import {
+  createAuthorization,
+  getAllResources,
+} from 'src/authorizations/actions/thunks'
+
+// Selectors
+import {getOrg} from 'src/organizations/selectors'
+import {getMe} from 'src/me/selectors'
+import {getAllTokensResources} from 'src/resources/selectors'
+
+// Helper Components
+import CodeSnippet from 'src/shared/components/CodeSnippet'
+
+// Utils
+import {allAccessPermissions} from 'src/authorizations/utils/permissions'
+import {event} from 'src/cloud/utils/reporting'
+
+// Types
+import {AppState, Authorization} from 'src/types'
+
+// Tokens stuff
+
+type OwnProps = {
+  wizardEventName: string
+  setTokenValue: (tokenValue: string) => void
+  tokenValue: string
+}
+
+const collator = new Intl.Collator(navigator.language || 'en-US')
+
+// Client stuff
+
+export const InitializeTokenClient: FC<OwnProps> = ({
+  wizardEventName,
+  setTokenValue,
+  tokenValue,
+}) => {
+  const org = useSelector(getOrg)
+  const me = useSelector(getMe)
+  const allPermissionTypes = useSelector(getAllTokensResources)
+  const dispatch = useDispatch()
+  const url =
+    me.quartzMe?.clusterHost || 'https://us-west-2-1.aws.cloud2.influxdata.com/'
+  const orgID = org.id
+  const currentAuth = useSelector((state: AppState) => {
+    return state.resources.tokens.currentAuth.item
+  })
+  const token = currentAuth.token
+
+  const codeSnippet = `influx config create --config-name onboarding \n
+  --host-url "${url}" \n
+  --org "${orgID}" \n
+  --token "${token}" \n
+  --active`
+
+  const bucketSnippet = `influx bucket create --name sample-bucket -c onboarding`
+
+  const sortedPermissionTypes = useMemo(
+    () => allPermissionTypes.sort((a, b) => collator.compare(a, b)),
+    [allPermissionTypes]
+  )
+
+  useEffect(() => {
+    const fetchResources = async () => {
+      await dispatch(getAllResources())
+    }
+    fetchResources()
+  }, [])
+
+  useEffect(() => {
+    if (sortedPermissionTypes.length && tokenValue === null) {
+      const authorization: Authorization = {
+        orgID: org.id,
+        description: `onboarding-${wizardEventName}-token-${Date.now()}`,
+        permissions: allAccessPermissions(sortedPermissionTypes, org.id, me.id),
+      }
+
+      dispatch(createAuthorization(authorization))
+      event(`firstMile.${wizardEventName}.tokens.tokenCreated`)
+    }
+  }, [sortedPermissionTypes.length])
+
+  // when token generated, save it to the parent component
+  useEffect(() => {
+    if (currentAuth.token) {
+      setTokenValue(currentAuth.token)
+    }
+  }, [currentAuth.token])
+
+  // Events log handling
+  const logCopyCodeSnippet = () => {
+    event(`firstMile.${wizardEventName}.tokens.code.copied`)
+  }
+
+  return (
+    <>
+      <h1>Initialize Client</h1>
+      <h2 style={{marginTop: '0px', marginBottom: '8px'}}>
```

Comment:
with these inline styles, if you find yoursleft making a lot of these same ones when making the next few onboarding components, let's create a stylesheet and pull them out into there.